File: example/javaexample/src/main/java/com/github/app/javaexample/ui/custom/CustomBackgroundIntro.java
package com.github.app.javaexample.ui.custom;

import android.graphics.Color;
import android.os.Bundle;

import androidx.fragment.app.Fragment;

import com.github.app.javaexample.R;
import com.github.appintro.AppIntro2;
import com.github.appintro.AppIntroFragment;


public class CustomBackgroundIntro extends AppIntro2 {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        addSlide(AppIntroFragment.newInstance(
                "Welcome!",
                "This is a demo of the AppIntro library, with a custom background on each slide!",
                 R.drawable.ic_slide1
        ));

        addSlide(AppIntroFragment.newInstance(
                "Clean App Intros",
                "This library offers developers the ability to add clean app intros at the start of their apps.",
                 R.drawable.ic_slide2
        ));

        addSlide(AppIntroFragment.newInstance(
                "Simple, yet Customizable",
                "The library offers a lot of customization, while keeping it simple for those that like simple.",
                 R.drawable.ic_slide3
        ));

        addSlide(AppIntroFragment.newInstance(
                "Explore",
                "Feel free to explore the rest of the library demo!",
                 R.drawable.ic_slide4
        ));

        // Set intro custom background
        setBackgroundResource(R.drawable.ic_drawer_header);

        // Change the color of the dot indicator.
        setIndicatorColor(Color.RED, Color.BLACK);
    }

    @Override
    protected void onDonePressed(Fragment currentFragment) {
        super.onDonePressed(currentFragment);
        finish();
    }

    @Override
    protected void onSkipPressed(Fragment currentFragment) {
        super.onSkipPressed(currentFragment);
        finish();
    }
}

File: example/src/main/java/com/github/appintro/example/ui/custom/CustomBackgroundIntro.kt
package com.github.appintro.example.ui.custom

import android.graphics.Color
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.github.appintro.AppIntro
import com.github.appintro.AppIntroBase
import com.github.appintro.SlidePageFragment
import com.github.appintro.example.R

class CustomBackgroundIntro : AppCompatActivity(), AppIntroBase.OnAppIntroListener {
    private lateinit var appIntro: AppIntro
    private val appIntroTag: String = "AppIntro"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_frame)

        if (savedInstanceState == null) {
            appIntro = AppIntro.newInstance(true)
            supportFragmentManager.beginTransaction()
                    .replace(R.id.app_intro_container, appIntro, appIntroTag)
                    .commitNow()
        } else {
            appIntro = supportFragmentManager.findFragmentByTag(appIntroTag) as AppIntro
        }
        appIntro.setAppIntroListener(this)
    }

    override fun onCreateAppIntro() {
        appIntro.addSlide(SlidePageFragment.newInstance(
                "Welcome!",
                "This is a demo of the AppIntro library, with a custom background on each slide!",
                imageDrawable = R.drawable.ic_slide1
        ))

        appIntro.addSlide(SlidePageFragment.newInstance(
                "Clean App Intros",
                "This library offers developers the ability to add clean app intros at the start of their apps.",
                imageDrawable = R.drawable.ic_slide2
        ))

        appIntro.addSlide(SlidePageFragment.newInstance(
                "Simple, yet Customizable",
                "The library offers a lot of customization, while keeping it simple for those that like simple.",
                imageDrawable = R.drawable.ic_slide3
        ))

        appIntro.addSlide(SlidePageFragment.newInstance(
                "Explore",
                "Feel free to explore the rest of the library demo!",
                imageDrawable = R.drawable.ic_slide4
        ))

        //set color for the next and skip arrow
        appIntro.setNextArrowColor(Color.RED)
        appIntro.setSkipArrowColor(Color.BLUE)

        // Set intro custom background
        appIntro.backgroundResource = R.drawable.ic_drawer_header

        // Change the color of the dot indicator.
        appIntro.setIndicatorColor(Color.RED, Color.BLACK)
    }

    override fun onBackPressed() {
        val used = appIntro.onBackPressed()
        if (!used) {
            return super.onBackPressed()
        }
    }

    override fun onSkipPressed(currentFragment: Fragment?) {
        finish()
    }

    override fun onDonePressed(currentFragment: Fragment?) {
        finish()
    }
}